modules/geoedgeRtdProvider.js

```diff
@@ -0,0 +1,238 @@
+function replacer(str) {
+  return function () {
+    return str;
+  }
+}
+
+export function wrapHtml(wrapper, html) {
+  return wrapper.replace(HTML_PLACEHOLDER, replacer(html));
+}
+
+/**
+ * generate macros dictionary from bid response
+ * @param {Object} bid
+ * @param {string} key
+ * @return {Object}
+ */
+function getMacros(bid, key = providerParams.key) {
+  return {
+    '${key}': key,
+    '%%ADUNIT%%': bid.adUnitCode,
+    '%%WIDTH%%': bid.width,
+    '%%HEIGHT%%': bid.height,
+    '%%PATTERN:hb_adid%%': bid.adId,
+    '%%PATTERN:hb_bidder%%': bid.bidderCode,
+    '%_isHb!': true,
+    '%_hbcid!': bid.creativeId || '',
+    '%%PATTERN:hb_pb%%': bid.pbHg,
+    '%%SITE%%': location.hostname,
+    '%_pimp%': PV_ID
+  };
+}
+
+/**
+ * replace macro placeholders in a string with values from a dictionary
+ * @param {string} wrapper
+ * @param {Object} macros
+ * @return {string}
+ */
+function replaceMacros(wrapper, macros) {
+  let key;
+  for (key in macros) {
+    wrapper = wrapper.replace(key, replacer(macros[key]));
+  }
+  return wrapper;
+}
+
+/**
+ * build final creative html with creative wrapper
+ * @param {Object} bid
+ * @param {string} wrapper
+ * @param {string} html
+ * @return {string}
+ */
+function buildHtml(bid, wrapper, html, key) {
+  let macros = getMacros(bid, key);
+  wrapper = replaceMacros(wrapper, macros);
+  return wrapHtml(wrapper, html);
+}
+
+/**
+ * muatates the bid ad property
+ * @param {Object} bid
+ * @param {string} ad
+ */
+function mutateBid(bid, ad) {
+  bid.ad = ad;
+}
+
+/**
+ * wraps a bid object with the creative wrapper
+ * @param {Object} bid
+ * @param {string} key
+ */
+export function wrapBidResponse(bid, key = providerParams.key) {
+  let wrapped = buildHtml(bid, wrapper, bid.ad, key);
+  mutateBid(bid, wrapped);
+}
+
+/**
+ * checks if bidder's bids should be monitored
+ * @param {string} bidder
+ * @return {boolean}
+ */
+function isSupportedBidder(bidder) {
+  return isEmpty(providerParams.bidders) || providerParams.bidders[bidder] === true;
+}
+
+/**
+ * checks if bid should be monitored
+ * @param {Object} bid
+ * @return {boolean}
+ */
+function shouldWrap(bid) {
+  let supportedBidder = isSupportedBidder(bid.bidderCode);
+  let wap = providerParams.wap ? preloaded : true;
+  return wrapperReady && supportedBidder && wap;
+}
+
+function conditionallyWrap(bid) {
+  if (shouldWrap(bid)) {
+    wrapBidResponse(bid);
+  }
+}
+
+function init(config, gdpr, usp) {
+  return true;
+}
+
+/** @type {RtdSubmodule} */
+export const geoedgeSubmodule = {
+  /**
+     * used to link submodule with realTimeData
+     * @type {string}
+     */
+  name: SUBMODULE_NAME,
+  /**
+     * get data and send back to realTimeData module
+     * @function
+     * @param {adUnit[]} adUnits
+     * @param {function} onDone
+     */
+  init,
+  onBidResponseEvent: conditionallyWrap
+};
+
+/**
+ * sets the module params from config
+ * @param {Object} realTimeData
+ */
+export function setParams(realTimeData) {
```

Comment:
This function is not needed. The RTD-core module will call you and pass config to conditionallyWrap().

You should be calling fetchWrapper() and preloadClient() in init().